gui/src/3D/materials/fluentButton/fluentButtonMaterial.ts

```diff
@@ -0,0 +1,494 @@
+import { Nullable } from "babylonjs/types";
+import { serializeAsColor4, serializeAsVector3, serialize, SerializationHelper } from "babylonjs/Misc/decorators";
+import { Matrix, Vector3, Vector4 } from "babylonjs/Maths/math.vector";
+import { IAnimatable } from "babylonjs/Animations/animatable.interface";
+import { BaseTexture } from "babylonjs/Materials/Textures/baseTexture";
+import { MaterialDefines } from "babylonjs/Materials/materialDefines";
+import { MaterialHelper } from "babylonjs/Materials/materialHelper";
+import { IEffectCreationOptions } from "babylonjs/Materials/effect";
+import { PushMaterial } from "babylonjs/Materials/pushMaterial";
+import { VertexBuffer } from "babylonjs/Meshes/buffer";
+import { AbstractMesh } from "babylonjs/Meshes/abstractMesh";
+import { SubMesh } from "babylonjs/Meshes/subMesh";
+import { Mesh } from "babylonjs/Meshes/mesh";
+import { Scene } from "babylonjs/scene";
+import { _TypeStore } from "babylonjs/Misc/typeStore";
+import { Color3, Color4 } from "babylonjs/Maths/math.color";
+import { EffectFallbacks } from "babylonjs/Materials/effectFallbacks";
+import { Constants, Texture } from "babylonjs";
+
+import { FluentGlowBlobTextureData } from "./fluentGlowBlob";
+
+import "./shaders/fluentButton.fragment";
+import "./shaders/fluentButton.vertex";
+
+/** @hidden */
+class FluentButtonMaterialDefines extends MaterialDefines {
+    public Relative_Width = true;
+    public Enable_Fade = true;
+    public _needNormals = true;
+    public _needUVs = true;
+
+    constructor() {
+        super();
+        this.rebuild();
+    }
+}
+
+/**
+ * Class used to render square buttons with fluent desgin
+ */
+export class FluentButtonMaterial extends PushMaterial {

```

Comment:
This file doesn't follow our naming convention - https://doc.babylonjs.com/divingDeeper/developWithBjs/approvedNamingConventions